lib/lumberjack/server.rb

```diff
@@ -37,36 +37,50 @@ def initialize(options={})
       # Query the port in case the port number is '0'
       # TCPServer#addr == [ address_family, port, address, address ]
       @port = @tcp_server.addr[1]
-      @ssl = OpenSSL::SSL::SSLContext.new
-      @ssl.cert = OpenSSL::X509::Certificate.new(File.read(@options[:ssl_certificate]))
-      @ssl.key = OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
+      @ssl_context = OpenSSL::SSL::SSLContext.new
+      @ssl_context.cert = OpenSSL::X509::Certificate.new(File.read(@options[:ssl_certificate]))
+      @ssl_context.key = OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
                                         @options[:ssl_key_passphrase])
-      @ssl_server = OpenSSL::SSL::SSLServer.new(@tcp_server, @ssl)
+      @stop = false
     end # def initialize
 
     def run(&block)
-      while true
-        connection = accept
-        Thread.new(connection) do |connection|
+      while running?
+        sock = @tcp_server.accept
```

Comment:
OpenSSL::SSL::SSLServer has a "start_immediately" property. This can be set to false to avoid the (potentially slow/blocking) handshake. Then in the client thread just call "sock.accept" to complete the handshake.

Might be cleaner and more intuitive to contributors than processing the socket creation yourself.